scipy/optimize/tests/test_optimize.py

```diff
@@ -2180,6 +2181,33 @@ def func(z, *params):
                           match=r'Either final optimization did not succeed'):
             optimize.brute(func, self.rranges, args=self.params, disp=True)
 
+        message = "Polishing is incompatible with integrality constraints"
+        with pytest.warns(OptimizeWarning, match=message):
+            optimize.brute(func, self.rranges, args=self.params,
+                           integrality=True)
+
```

Comment:
No test for what happens if there are no integers between `(low, high)`.

Reply:
In `main`, when there are no values in the slice:
```python3
res = brute(f, [slice(10, 10, 10), (20, 10)], finish=False)
```
we get
```

Traceback (most recent call last):

  File "C:\Users\matth\scipy\scipy\optimize\tests\untitled0.py", line 48, in <module>
    res = brute(f, [slice(10, 10, 10), (20, 10)], finish=False)

  File "C:\Users\matth\scipy\scipy\optimize\_optimize.py", line 3506, in brute
    indx = argmin(Jout.ravel(), axis=-1)

  File "<__array_function__ internals>", line 5, in argmin

  File "C:\Users\matth\Anaconda3\envs\scipydev\lib\site-packages\numpy\core\fromnumeric.py", line 1276, in argmin
    return _wrapfunc(a, 'argmin', axis=axis, out=out)

  File "C:\Users\matth\Anaconda3\envs\scipydev\lib\site-packages\numpy\core\fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)

ValueError: attempt to get argmin of an empty sequence
```